### DISCUSSION TITLE
#11 - Extra sussy refactoring recap ඞ

### DISCUSSION THREAD

**Discussion by jas20202** [CUMGroup/AMOGUS#86] [Blog]
Hällo there fellow AMOGUS fans uwu,

we were a bit ahead of our time last week and had already done a lot of refactoring 👀
We mainly focused on abstracting more functionality away with their own interfaces, so that testing will be a lot easier in the future.
Additionally we cleaned up a lot of methods, which seemed to long and incohesive. 
Our backend code-base is much cleaner now!🧹💉
You can find a more detailed description of the refactoring here: [AMOGUS DOCS](https://cumgroup.github.io/AMOGUS/Developer%20Docs/Architecture/Backend/Refactor/).

Furthermore we made some improvements in the front-end. @WachtelHD is currently working on the teacher view as well as improving the game view, meanwhile @Annalytic-programming is working on our landing page connecting it to the game and the how-to-page.
Finally, @jas20202 has worked on improving the stats view and the login page and @AlexMi-Ha is currently implementing all needed Services in the front-end. 

We still have a handful of things to work on but we are starting to see the finished line in the horizon 🏃‍♂️🏃‍♂️🏃‍♀️🏃‍♀️

To round off this blog post here are our reworked pages so far:
![image](https://user-images.githubusercontent.com/99250573/233837275-9687bff1-c00b-4b89-a14e-6190ce1e8212.png)
![image](https://user-images.githubusercontent.com/99250573/233837590-dae86735-a2da-4324-80db-8a1e5baa7e3d.png)

All being said, never forget, always stay sus ඞ

Yours truly,
CUMGroup 🚗

**Comment by DanielSchirmer** [comment]
Hello Team Amogus,
good to hear that you are already thinking about testing and adding functionality to make it easier later. The sooner the better! Did you use the code principles KISS, DRY or YAGNI for the cleanup of your source code? If not, you might want to take a look at them. Furthermore your frontend looks very appealing. Keep it up!
~GuessMaster

**Reply by AlexMi-Ha** [reply]
Hello Team GuessMaster,
Yes, certainly! We tried to apply various state-of-the-art principles to improve our codebases maintainability, readability, testability and so on. KISS, DRY and YAGNI can't be missing there of course 😉

Yours truly,
CUMGroup

**Reply by Julian702** [reply]
Hello Team Amogus,

Great to hear that you already thought ahead and did some refactoring. As you mentioned that you abstracted to improve testability in the future, we were wondering if you already have a plan on what testing framework you want to utilize?

We are looking forward for your projects development.

Your Kioku Team

**Reply by AlexMi-Ha** [reply]
Hi Julian,
we already thought about this and are taking the obvious choice for C#/.NET. We will be using the [xUnit](https://github.com/xunit/xunit) testing framework for Unit Testing together with the [Moq](https://github.com/moq/moq4) library for mocking the dependencies.
Additionally, for Type/IO Validation we will be using the [FluentValidation](https://github.com/FluentValidation/FluentValidation) library.

Yours truly
CUMGroup

**Comment by Luca-Hackl** [comment]
Hello Team Amogus,

It’s good to hear that you already started with refactoring! The only thing we as RALF would are missing is a few examples how you refactored the code exactly. 

Yours truly,
Ralf
